protocol/src/lib.rs

```diff
@@ -197,6 +197,10 @@ pub enum Message {
     ReadRequest(Uuid, u64, Vec<u64>, Vec<ReadRequest>),
     ReadResponse(Uuid, u64, Result<Vec<ReadResponse>, CrucibleError>),
 
+    // ReadWithFill: Uuid, job id, dependencies, [ReadRequest], [Write]
+    ReadFill(Uuid, u64, Vec<u64>, Vec<Write>),
```

Comment:
this is missing `Vec<ReadRequest>`?

Reply:
ignore this - we discussed my confusion here, and this isn't a Read

Reply:
Side channel discussion and rename to WriteUnwritten, so it's clear we don't need a `ReadRequest` here